### PR TITLE
WFLY-11436 For WildFly 16, also remove NON_XA from WildFly high availability configuration

### PR DESCRIPTION
**Pull request by scottmarlow**
https://issues.jboss.org/browse/WFLY-11436

### CODE REVIEW

File: clustering/infinispan/extension/src/main/resources/subsystem-templates/infinispan.xml [hunk 1]
```diff
@@ -70,7 +70,6 @@
                     <expiration max-idle="100000"/>
                 </local-cache>
                 <invalidation-cache name="entity">
-                    <transaction mode="NON_XA"/>
```
Comment (pferraro) [inline clustering/infinispan/extension/src/main/resources/subsystem-templates/infinispan.xml]: This isn't sufficient to affect the default configuration.  You need to modify the following files:
* galleon-pack/src/main/resources/feature_groups/infinispan-dist.xml
* galleon-pack/src/main/resources/feature_groups/infinispan-local.xml

**Review by pferraro — CHANGES_REQUESTED** [review]

File: clustering/infinispan/extension/src/main/resources/subsystem-templates/infinispan.xml (hunk 1 above)
Comment (kabir) [reply clustering/infinispan/extension/src/main/resources/subsystem-templates/infinispan.xml]: @scottmarlow ^^

Comment (scottmarlow) [reply clustering/infinispan/extension/src/main/resources/subsystem-templates/infinispan.xml]: Thanks @pferraro!  Will make that change also.  Thanks @kabir for the ping :)

Comment (scottmarlow) [reply clustering/infinispan/extension/src/main/resources/subsystem-templates/infinispan.xml]: @pferraro please review my update to galleon-pack/src/main/resources/feature_groups/infinispan-dist.xml + galleon-pack/src/main/resources/feature_groups/infinispan-local.xml, to make sure that I didn't remove too much.

**Review by pferraro — APPROVED** [review]

File: clustering/infinispan/extension/src/main/resources/subsystem-templates/infinispan.xml (hunk 1 above)
Comment (pferraro) [reply clustering/infinispan/extension/src/main/resources/subsystem-templates/infinispan.xml]: Looks good now.

**Pull request merged**